daemon/containerd/image_children.go

```diff
@@ -127,3 +128,113 @@ func isRootfsChildOf(child ocispec.RootFS, parent ocispec.RootFS) bool {
 
 	return true
 }
+
+// Parents returns a slice of image IDs whose entire rootfs contents match,
+// in order, the childs first layers, excluding images with the exact same
+// rootfs.
+// The returned slice is sorted in descending order by affinity, meaning
+// closer parents (images whose rootfs have more matching layers) come first
+// and more distant relatives come last.
+// TODO(laurazard): maybe this could take an optional sort by func?
```

Comment:
Right, no additional round trips was why I suggested letting the caller pass a sort function in. Having parents return all the necessary information for the caller to sort it themselves works too.

Reply:
Returning all necessary information to the caller [also allows the caller to avoid `i.resolveImage()` _after_ sorting, too.](https://github.com/moby/moby/pull/45427/files#diff-5d5a46c403a4d9666b17b889c12655b3c1539ab462cc38f3fd2e43cf99749b43R136-R137)

Reply:
D'oh, I read your first message too quickly and didn't realize that's what you meant. Sounds good, although it might seem a bit weird to have `Children` return a slice of imageIDs and `Parents` return something else. We really should standardize a little bit around what we're returning/passing around from these functions, but I guess that's easier to figure out after we've implemented some of it.

Reply:
Any suggestions for alternate return types here? Nothing immediately jumps out at me (but that is probably due to my limited knowledge of the codebase/containerd types), other than separately returning a slice of `(containerd)images.Image` and a `map[(containerdimages.Image.Name)][]ocispec.Rootfs`, but not sure that's optimal. cc @vvoland @rumpl @thaJeztah

Reply:
Since it's now a private method and therefore an implementation detail, I figure just define a new unexported struct type to return a slice of.